rbac_api.go

```diff
@@ -171,14 +171,25 @@ func (e *Enforcer) GetImplicitRolesForUser(name string, domain ...string) []stri
 //
 // GetPermissionsForUser("alice") can only get: [["alice", "data2", "read"]].
 // But GetImplicitPermissionsForUser("alice") will get: [["admin", "data1", "read"], ["alice", "data2", "read"]].
-func (e *Enforcer) GetImplicitPermissionsForUser(user string) [][]string {
-	roles := e.GetImplicitRolesForUser(user)
-	roles = append([]string{user}, roles...)
-
+func (e *Enforcer) GetImplicitPermissionsForUser(user string, domain ...string) [][]string {
+	roles := e.GetImplicitRolesForUser(user, domain...)
 	res := [][]string{}
+	permissions := [][]string{}
+
 	for _, role := range roles {
-		permissions := e.GetPermissionsForUser(role)
+		permissions = e.GetPermissionsForUser(role)
 		res = append(res, permissions...)
 	}
+
+	if len(domain) == 1 {
+		permissions = e.GetPermissionsForUserInDomain(user, domain[0])
```

Comment:
Why only retrieve the permissions for ``user``, what about ``roles``?

Reply:
Check https://github.com/casbin/casbin/pull/217/files/6d2f0eef4c310a0446a74b7429f9ff6448c4a076#diff-46334e8bf05954874b90372da400c43fR179